src/dataflow-types/src/types.rs

```diff
@@ -93,6 +93,17 @@ pub struct SourceInstanceDesc {
     pub persist: Option<sources::persistence::SourcePersistDesc>,
 }
 
+/// A representation of `SourceInstanceDesc` which elides the source details.
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Serialize, Deserialize)]
+pub struct SourceInstanceKey {
```

Comment:
Add a `fn` to construct from a `GlobalId` and a `SourceInstanceDesc`? This would save some typing later.

Reply:
Yup! Will do.

Reply:
Fixed!